lib/i18n/tokenizer/base.ts

```diff
@@ -199,7 +202,10 @@ export default class BaseTokenizer {
         // numbers are considered part of a word if preceded by a letter
         // if a word ends with hyphen followed by a number, the hyphen is considered part of the word: e.g., top-50 -> top- 50
         // if a word ends with hyphen, the hyphen is not considered part of the word: e.g., Twitter- -> Twitter -
-        this._addDefinition('WORD', /(?:-?{LETTER}[0-9۰-۹]*)+(?:-?(?=[0-9۰-۹]))?/);
+        // updates:
+        // ... except for -lrb- and -rrb-
+        // person first names or abbreviations (such as st.) may end with a dot; do not split them
```

Comment:
These commits are mainly the result of my experience with typechecking SPL and autoQA datasets to port them to thingtalk2. 
If the sentence contains abbreviations like "st.", "dr.", etc. they won't typecheck.
It might be better to not split them unless they are period at the end of a sentence?

Reply:
Yes the issue of course is that you have no idea if something is a period at the end of the sentence or not, if a single command is made of two sentences!

Reply:
Actually, I just noticed we have _initAbbrv() and those tokens should go there...

Reply:
Yes if you want to extend the list of abbreviations for English (without making every word followed by . into an abbreviation). But I tried to keep the list smaller than the old CoreNLP-based tokenizer because the more we do the more we risk making mistakes.